### PR TITLE
feat: custom snackbar

### LINKED ISSUE #3096: [Demand] Custom notistack for ethereum transaction
## Catalogue

- [x] UI/UX
- [ ] Mask Core Feature
- [ ] Mask Plugin Feature
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background

Now the content of the toast is just one sentence

## Part 3: Expected Behavior

Now we need an ethereum transaction notistack which has a title and a description, and a tx link at the end of the description.

## Notes

Figma: https://www.figma.com/file/gVkQ67y285b4FXVV1KPThN/Twitter?node-id=11154%3A21
notistack doc: https://iamhosseindhv.com/notistack/demos#custom-snackbar

### PR DESCRIPTION
**Pull request by zhouhanseng**
closes #3096

image

### CODE REVIEW

File: packages/maskbook/src/plugins/Snapshot/apis/index.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export async function vote(identifier: ProposalIdentifier, choice: number, addre
         },
     })
 
-    const sig = await Services.Ethereum.sign(msg, address)
+    const sig = await Services.Ethereum.personalSign(msg, address)
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Snapshot/apis/index.ts:69]: cc @guanbinrui

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

File: packages/maskbook/src/plugins/Snapshot/apis/index.ts (hunk 1 above)
Comment (zhouhanseng) [reply packages/maskbook/src/plugins/Snapshot/apis/index.ts:69]: I found that `sign()` does not awake metamask sign interface window well.

File: packages/maskbook/src/plugins/Wallet/UI/WalletStatusDialog.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import ErrorIcon from '@material-ui/icons/Error'
 import { Button, DialogActions, DialogContent, Link, makeStyles, Typography } from '@material-ui/core'
 import { InjectedDialog } from '../../../components/shared/InjectedDialog'
 import { ProviderIcon } from '../../../components/shared/ProviderIcon'
-import { useSnackbarCallback } from '../../../extension/options-page/DashboardDialogs/Base'
+import { useSnackbarCallback } from '../../../utils'
 import Services from '../../../extension/service'
 import { useRemoteControlledDialog, useValueRef, useI18N } from '../../../utils'
```
Comment (UncleBill) [inline packages/maskbook/src/plugins/Wallet/UI/WalletStatusDialog.tsx:10]:
```suggestion
import { useSnackbarCallback, useRemoteControlledDialog, useValueRef, useI18N } from '../../../utils'
import Services from '../../../extension/service'
```

**Review by UncleBill — COMMENTED** [review]
Also please handle these warnings
image

**Comment by septs** [comment]
@zhouhanseng plz delete branch?

**Pull request closed without merging**